### ISSUE TITLE
Support macOS Big Sur

### ISSUE THREAD

**Issue by liam-ly** [Simspace/ld-wrapper-macos#6]
Out of the box, the test package fails on macOS Big Sur:
```
stack build panic
ld: library not found for -liconv
```
I tried to do
```
brew install libiconv
LDFLAGS="-L/usr/local/opt/libiconv/lib" stack build panic
```
but there was no change.

This is likely https://gitlab.haskell.org/ghc/ghc/-/issues/18446

If the workaround mentioned in the above ticket is going to work, I suspect changes are needed in this repo and the workaround.

**Comment by jship** [comment]
Thanks @liam-ly for the report.

> This is likely https://gitlab.haskell.org/ghc/ghc/-/issues/18446

Yes, it appears we're stuck between a rock and hard place here for the time being until updated versions of GHC are released containing the fix addressing 18446. Merges landed about a month ago for both [8.8.5](https://gitlab.haskell.org/ghc/ghc/-/merge_requests/4455) and [8.10.3](https://gitlab.haskell.org/ghc/ghc/-/merge_requests/4462).

I take it you are running a GHC version prior to 8.6.1 in your projects?  SimSpace encountered this issue heavily when we were running GHC 8.2, but we haven't needed to leverage the wrapper script as of switching to 8.8.  8.6.1 should be where the GHC-side fix went in that obviated the need for the wrapper script for us.

**Comment by liam-ly** [comment]
> Merges landed about a month ago

Yep, this was me :) 

> but we haven't needed to leverage the wrapper script as of switching to 8.8. 8.6.1 should be where the GHC-side fix went in that obviated the need for the wrapper script for us.

Oh! I will try this out. I haven't re-tested since we upgraded past 8.6.1. I will report back on the results.

**Comment by liam-ly** [comment]
Indeed, changing the snapshot to 16.20 (ghc-8.8.4) compiles on Big Sur without the wrapper script. I'm unblocked, but I will leave it up to you @jship if this ticket should be closed or not since it's blocked on 18446 being backported and released in 8.6, or someone gets https://github.com/yairchu/macos11-haskell-workaround working along with this script.
